File: javascript/gpt/p113.js
// p113.js - check if host is alive via ping
const { spawnSync } = require('child_process');
const host = process.argv[2];
if (!host) { console.error('usage: node p113.js host'); process.exit(1); }
const res = spawnSync('ping', ['-c','1', host]);
if (res.status === 0) console.log('alive'); else console.log('not alive');
process.exit(res.status);

File: python/gpt/p113.py
"""
p113.py
Run ping and return 'alive' if returncode == 0 else 'not alive'. CLI: python p113.py host
"""
import sys
import subprocess

if len(sys.argv) < 2:
    print('usage: p113.py host')
    sys.exit(1)
host = sys.argv[1]
try:
    res = subprocess.run(['ping','-c','1', host], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
    print('alive' if res.returncode == 0 else 'not alive')
    sys.exit(res.returncode)
except FileNotFoundError:
    print('ping not found')
    sys.exit(2)
